Unit test failure (`test_version`) (v0.4.1)

**Describe the bug**

Failed `TestVersion.test_version` when running comprehensive test suite.

**To Reproduce**

`cd MontePy`
`pytest`

**Error Message (if any)**
```
================================================== FAILURES ==================================================
__________________________________________ TestVersion.test_version __________________________________________

self = <tests.test_version.TestVersion testMethod=test_version>

    def test_version(self):
        print(os.path.join("montepy", "_version.py"))
>       self.assertTrue(os.path.exists(os.path.join("montepy", "_version.py")))
E       AssertionError: False is not true

tests/test_version.py:10: AssertionError
-------------------------------------------- Captured stdout call --------------------------------------------
montepy/_version.py
========================================== short test summary info ===========================================
FAILED tests/test_version.py::TestVersion::test_version - AssertionError: False is not true
```

**Version**

 - Version 0.4.1 with optional testing dependencies (`pip install 'montepy[test]'`)
 - Python 3.12.3

**Additional context**

[Part of pyOpenSci review!](https://github.com/pyOpenSci/software-submission/issues/205)

--- a/tests/test_version.py
+++ b/tests/test_version.py
@@ -1,9 +1,16 @@
 import importlib
 import os
+import subprocess
 import sys
+from pathlib import Path
 
 
 def test_version():
+    version_file = Path("montepy") / "_version.py"
+    if not version_file.exists():
+        subprocess.run(
+            ["python", "-m", "setuptools_scm", "--force-write-version-files"]
+        )
     assert os.path.exists(os.path.join("montepy", "_version.py"))
     # Test without version.py
     try:

--- a/tests/test_version.py
+++ b/tests/test_version.py
@@ -1,38 +1,36 @@
 import importlib
 import os
 import sys
-from unittest import TestCase
 
 
-class TestVersion(TestCase):
-    def test_version(self):
-        self.assertTrue(os.path.exists(os.path.join("montepy", "_version.py")))
-        # Test without version.py
-        try:
-            # try without setuptools_scm
-            old_file = os.path.join("montepy", "_version.py")
-            new_file = os.path.join("montepy", "_version.bak")
-            os.rename(old_file, new_file)
-            # clear out previous imports
-            to_delete = set()
-            for mod in sys.modules:
-                for bad_mod in ["setuptools_scm", "montepy"]:
-                    if bad_mod in mod:
-                        to_delete.add(mod)
-            for mod in to_delete:
-                del sys.modules[mod]
-            sys.modules["setuptools_scm"] = None
-            import montepy
+def test_version():
+    assert os.path.exists(os.path.join("montepy", "_version.py"))
+    # Test without version.py
+    try:
+        # try without setuptools_scm
+        old_file = os.path.join("montepy", "_version.py")
+        new_file = os.path.join("montepy", "_version.bak")
+        os.rename(old_file, new_file)
+        # clear out previous imports
+        to_delete = set()
+        for mod in sys.modules:
+            for bad_mod in ["setuptools_scm", "montepy"]:
+                if bad_mod in mod:
+                    to_delete.add(mod)
+        for mod in to_delete:
+            del sys.modules[mod]
+        sys.modules["setuptools_scm"] = None
+        import montepy
 
-            self.assertEqual(montepy.__version__, "Undefined")
-            # try with setuptools_scm
-            del sys.modules["setuptools_scm"]
-            importlib.reload(montepy)
-            print(f"From setuptools_scm: {montepy.__version__}")
-            self.assertTrue(len(montepy.__version__.split(".")) >= 3)
-        finally:
-            os.rename(new_file, old_file)
-        # do base with _version
+        assert montepy.__version__ == "Undefined"
+        # try with setuptools_scm
+        del sys.modules["setuptools_scm"]
         importlib.reload(montepy)
-        print(f"From _version.py: {montepy.__version__}")
-        self.assertTrue(len(montepy.__version__.split(".")) >= 3)
+        print(f"From setuptools_scm: {montepy.__version__}")
+        assert len(montepy.__version__.split(".")) >= 3
+    finally:
+        os.rename(new_file, old_file)
+    # do base with _version
+    importlib.reload(montepy)
+    print(f"From _version.py: {montepy.__version__}")
+    assert len(montepy.__version__.split(".")) >= 3

